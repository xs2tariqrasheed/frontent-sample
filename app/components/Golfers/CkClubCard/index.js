import React from 'react';
import PropTypes from 'prop-types';
import GolferImage from 'components/GolferImage';
import {
  Container,
  ImageContainer,
  ImageWrapper,
  ImageScaler,
  FounderImage,
  ImageFullWrap,
} from './styled';
import FounderStamp from './FounderStamp.svg';

const CkClubCard = props => {
  const playerType = (power, accuracy, composure, stamina) => {
    if (power >= accuracy && power >= stamina && power >= composure) {
      return { status: 'Power Player', color: '#e6fff2' };
    }
    if (accuracy >= power && accuracy >= stamina && accuracy >= composure) {
      return { status: 'Accuracy Player', color: '#e6f9ff' };
    }
    if (composure >= power && composure >= stamina && composure >= accuracy) {
      return { status: 'Composure Player', color: '#fff9e6' };
    }

    return { status: 'Stamina Player', color: '#f3ffe6' };
  };

  const statusColor = playerType(
    +props.golfer.powerPeak,
    +props.golfer.accuracyPeak,
    +props.golfer.composurePeak,
    +props.golfer.staminaPeak,
  );

  return (
    <Container>
      <ImageContainer color={statusColor.color}>
        <ImageScaler>
          <FounderImage src={FounderStamp} alt="FounderTag" />
          <ImageWrapper {...props}>
            <ImageFullWrap>
              <GolferImage
                image_url={props.golfer.image_url}
              />
            </ImageFullWrap>
          </ImageWrapper>
        </ImageScaler>
      </ImageContainer>
    </Container>
  );
};

CkClubCard.propTypes = {
  golfer: PropTypes.object,
};

export default CkClubCard;
