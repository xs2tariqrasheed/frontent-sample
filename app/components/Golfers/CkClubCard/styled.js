import styled, { css } from 'styled-components';
import Img from 'components/Img';

export const Container = styled.div`
  width: 320px;
  height: auto;
  background-color: #ffffff;
  margin-left: 12px;
  margin-right: 12px;
  margin-bottom: 12px;
  margin-top: 12px;
`;

export const ImageScaler = styled.div`
  transform: scale(1.2, 1.2);
  margin-left: 18px;
`;

export const FounderImage = styled(Img)`
  position: absolute;
  transform: translate(40px, 20px);
`;

export const ImageFullWrap = styled.div`
  transform: scale(0.35, 0.35);
  width: 188px;
  height: 188px;
  object-fit: contain;
`;

export const ImageWrapper = styled.div`
  position: relative;

  ${props =>
    props.golfer.bodyShape === 1 &&
    css`
      transform: translate(10px, -30px);
    `}

  ${props =>
    props.golfer.bodyShape === 2 &&
    css`
      transform: translate(20px, -30px);
    `}

  ${props =>
    props.golfer.bodyShape === 3 &&
    css`
      transform: translate(15px, -30px);
    `}
    
`;

export const ImageContainer = styled.div`
  position: relative;
  width: 272px;
  height: 213px;
  margin-top: 24px;
  margin-left: 24px;
  border-radius: 5px;
  background-color: ${props => props.color};
  clip-path: inset(0px 0px 0px 0px);
`;

export const Oval = styled.div`
  margin-left: 65px;
  border-radius: 50%;
  width: 146px;
  height: 18px;
  opacity: 0.35;
  mix-blend-mode: multiply;
  background-color: #d8d8d8;
`;
