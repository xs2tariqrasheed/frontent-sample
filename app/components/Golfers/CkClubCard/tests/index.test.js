import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../../configureStore';
import CkClubCard from '../index';

describe('<CkClubCard />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store.store}>
        <IntlProvider locale="en">
          <CkClubCard
            golfer={{
              tokenid: 1,
              powertotal: 30,
              accuracytotal: 71,
              composuretotal: 43,
              staminatotal: 99,
              powerpeak: 30,
              accuracypeak: 71,
              composurepeak: 43,
              staminapeak: 99,
              sex: 0,
              driverIndex: 1,
              skinToneIndex: 1,
              eyeIndex: 1,
              noseIndex: 1,
              mouthIndex: 1,
              eyebrowIndex: 1,
              hairIndex: 1,
              hatIndex: 1,
              pantsIndex: 1,
              shirtIndex: 1,
              shoeIndex: 1,
            }}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
