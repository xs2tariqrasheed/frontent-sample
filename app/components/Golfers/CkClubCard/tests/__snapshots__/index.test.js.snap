// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CkClubCard /> should render and match the snapshot 1`] = `
<div
  class="styled__Container-tq8q6d-0 iXdWjj"
>
  <div
    class="styled__ImageContainer-tq8q6d-5 iOBioD"
    color="#f3ffe6"
  >
    <div
      class="styled__ImageScaler-tq8q6d-1 dHohDI"
    >
      <img
        alt="FounderTag"
        class="styled__FounderImage-tq8q6d-2 ivJQMe"
        src="IMAGE_MOCK"
      />
      <div
        class="styled__ImageWrapper-tq8q6d-4 liPFjY"
      >
        <div
          class="styled__ImageFullWrap-tq8q6d-3 kWEmjd"
        >
          <div
            class="relative h-full"
            style="min-width: 200px;"
          >
            <img
              alt="body"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="head"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="eyebrow"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="eyes"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="mouth"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="hair"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="shirt"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="pants"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="shoe"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="driver"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="nose"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
            <img
              alt="hat"
              class="styled__Avatar-sc-1g9vwal-1 gulTga"
              src="IMAGE_MOCK"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
